Marlin/src/libs/autoreport.h

```diff
@@ -25,23 +25,25 @@
 
 template <typename Helper>
 struct AutoReporter {
-  millis_t next_report_ms;
+  millis_t last_report_ms;
   uint8_t report_interval;
+  millis_t report_interval_ms;
   #if HAS_MULTI_SERIAL
     SerialMask report_port_mask;
     AutoReporter() : report_port_mask(SerialMask::All) {}
   #endif
 
   inline void set_interval(uint8_t seconds, const uint8_t limit=60) {
     report_interval = _MIN(seconds, limit);
-    next_report_ms = millis() + SEC_TO_MS(seconds);
+    last_report_ms=millis();
+    report_interval_ms=SEC_TO_MS(report_interval); //pre calculate interval in milliseconds
   }
 
   inline void tick() {
     if (!report_interval) return;
     const millis_t ms = millis();
-    if (ELAPSED(ms, next_report_ms)) {
-      next_report_ms = ms + SEC_TO_MS(report_interval);
+    if (millis_t(ms - last_report_ms) > report_interval_ms) {
```

Comment:
```
millis_t(ms - last_report_ms) > report_interval_ms
```
… is essentially equivalent to …
```
int32_t(ms - last_report_ms - report_interval_ms) > 0
```
…for any `report_interval_ms` value less than 24.85 days. And `report_interval_ms` has a maximum value of 60,000 ms.

It doesn't matter if `millis()` is about to roll over. The result will be exactly the same.

Reply:
Ok, yeah. This is probably is so. I take it that You refer to me not using PENDING macro here.
`millis_t(ms - last_report_ms) > report_interval_ms`
 To me it seems so much more readable, that the interval is on one side of the comparison, it instantly makes it clear that I am dealing with interval timer.
 Versus `int32_t(ms - last_report_ms - report_interval_ms) > 0` that takes some time to figure out. Both assumably compile to the exact same.

> ```
> millis_t(ms - last_report_ms) > report_interval_ms
> ```
> 
> … is essentially equivalent to …
> 
> ```
> int32_t(ms - last_report_ms - report_interval_ms) > 0
> ```
> 
> …for any `report_interval_ms` value less than 24.85 days. And `report_interval_ms` has a maximum value of 60,000 ms.
> 
> It doesn't matter if `millis()` is about to roll over. The result will be exactly the same.

Reply:
Ouch just noticed ` int32_t` in there: `int32_t(ms - last_report_ms - report_interval_ms) > 0`
This is a SIGNED so it do not work exactly the same. Would it be `uint32_t` UNSIGNED it would be equivalent. Because it is assumed millis_t is unsigned(as it currently is).